app/src/main/java/org/groundplatform/android/MainActivity.kt

```diff
@@ -234,4 +252,11 @@ class MainActivity : AbstractActivity() {
       navHostFragment.navController.navigate(directions)
     }
   }
+
+  private fun NavController.handleDeepLinkIfNeeded(uri: Uri) {
+    if (uri.pathSegments.firstOrNull() == SURVEY_PATH_SEGMENT) {
+      val surveyId = uri.lastPathSegment
+      navigate(SurveySelectorFragmentDirections.showSurveySelectorScreen(false, surveyId))
```

Comment:
As discussed previously, why are we using `survey selector fragment` for this flow? Is this a UX requirement?

Reply:
Discussed in webdnesday meeting, showed the flow, we can do this for now.